backend/app/adapter/db/user_changelog.go

```diff
@@ -0,0 +1,89 @@
+package db
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/short-d/short/app/adapter/db/table"
+	"github.com/short-d/short/app/entity"
+	"github.com/short-d/short/app/usecase/repository"
+)
+
+var _ repository.UserChangeLog = (*UserChangeLogSQL)(nil)
+
+// UserChangeLogSQL accesses UserChangeLog information in user_changelog table through SQL.
+type UserChangeLogSQL struct {
+	db *sql.DB
+}
+
+// GetLastViewedAt finds LastViewedAt time for given user.
+func (u UserChangeLogSQL) GetLastViewedAt(user entity.User) (time.Time, error) {
+	statement := fmt.Sprintf(`
+SELECT "%s" 
+FROM "%s"
+WHERE "%s"=$1;`,
+		table.UserChangeLog.ColumnLastViewedAt,
+		table.UserChangeLog.TableName,
+		table.UserChangeLog.ColumnEmail,
+	)
+
+	row := u.db.QueryRow(statement, user.Email)
+	lastViewedAt := time.Time{}
+	err := row.Scan(&lastViewedAt)
+	return lastViewedAt, err
+}
+
+// UpdateLastViewedAt updates LastViewedAt time for given user to currentTime.
+func (u UserChangeLogSQL) UpdateLastViewedAt(user entity.User, currentTime time.Time) (time.Time, error) {
+	lastViewedAt, err := u.GetLastViewedAt(user)
```

Comment:
For me this check is redundant.  We can check how many rows were affected by `UPDATE` query and reduce the number of Postgres calls.